.ci/test_r_package.sh

```diff
@@ -98,7 +98,7 @@ if grep -q -R "WARNING" "$LOG_FILE_NAME"; then
     exit -1
 fi
 
-ALLOWED_CHECK_NOTES=2
+ALLOWED_CHECK_NOTES=4
```

Comment:
@jameslamb Should be 3, right?

Reply:
yes  @btrotta can you please change to 3?

A lot has changed in our CI since this PR was first opened, sorry 😬

Reply:
```suggestion
ALLOWED_CHECK_NOTES=3
```

Reply:
@jameslamb Is a similar increment needed for Windows?
https://github.com/microsoft/LightGBM/blob/2c18a0f3ed732df77d4304ed1d2b46a345d661c8/.ci/test_r_package_windows.ps1#L102

Reply:
No worries, fixed now.

Reply:
ping @jameslamb for Windows updates

Reply:
@StrikerRUS nothing is required on this PR since [the Windows R tests are passing](https://ci.appveyor.com/project/guolinke/lightgbm/builds/32470697/job/lkhm0f63w8titj1w).  We have it set to 3  and are generating exactly  3:

![image](https://user-images.githubusercontent.com/7608904/80558365-c23ffc00-899f-11ea-8d0a-0eadb0fe82fe.png)

Reply:
@jameslamb OK. I just was afraid that we can hit https://github.com/microsoft/LightGBM/pull/2950#issuecomment-605696533 on Windows if let say the current size is 4.99 during some runs it can be calculated as 5, like in #2988.

Reply:
If we do we can bump it back up and it wouldn't bother me. I just don't think this PR has to be the place where we deal with it. If CI is passing for the R package right now, nothing should be changed.